app/models/book.rb

```diff
@@ -33,9 +33,8 @@ class Book < ActiveRecord::Base
   }
 
   def calculate_avg_score
-    size = ratings.size
-    sum = ratings.inject(0.0) { |accum, rating| accum + rating.score }
-    update_attributes(avg_score: sum / size)
+    avg = ratings.average(:score)
```

Comment:
use .to_f add the end, average can return nil, with to_f nil is turned into a zero